packages/hdwallet-tallyho/src/tallyho.ts

```diff
@@ -222,6 +280,59 @@ export class TallyHoHDWallet implements core.HDWallet, core.ETHWallet {
     return false;
   }
 
+  private async detectTallyProvider(): Promise<TallyHoEthereumProvider | null> {
+    let handled = false;
+
+    return new Promise((resolve) => {
+      if ((window as Window).ethereum) {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        handleEthereum();
+      } else {
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        window.addEventListener("ethereum#initialized", handleEthereum, { once: true });
+
+        setTimeout(() => {
+          // eslint-disable-next-line @typescript-eslint/no-use-before-define
+          handleEthereum();
+        }, 3000);
+      }
+
+      function handleEthereum() {
+        if (handled) {
+          return;
+        }
+        handled = true;
+
+        window.removeEventListener("ethereum#initialized", handleEthereum);
+
+        const { ethereum } = window as Window;
+
+        if (ethereum && ethereum.isTally) {
+          resolve(ethereum as unknown as TallyHoEthereumProvider);
+        } else {
+          const message = ethereum ? "Non-TallyHo window.ethereum detected." : "Unable to detect window.ethereum.";
+
+          console.error("hdwallet-tallyho: ", message);
+          resolve(null);
+        }
+      }
+    });
+  }
+  public async ethSwitchChain(chainId = 1): Promise<void> {
+    const hexChainId = ethers.utils.hexValue(chainId);
+    try {
+      // at this point, we know that we're in the context of a valid MetaMask provider
+      const provider: any = await this.detectTallyProvider();
+      await provider.request({ method: "wallet_switchEthereumChain", params: [{ chainId: hexChainId }] });
+    } catch (e: any) {
+      const error: core.SerializedEthereumRpcError = e;
+      if (error.code === 4902) {
+        // TODO: EVM Chains Milestone
+        // We will need to pass chainName and rpcUrls, which we don't have yet, to add a chain to Tally.
+      }
```

Comment:
Log the error for both the 4902-specific and all other cases to ease debugging.

Reply:
Added a console.error in https://github.com/shapeshift/hdwallet/commit/de254daa059182713ac2e6c2d17869ff1fc55840.